src/shims/intrinsics.rs

```diff
@@ -223,6 +223,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let discr_val = this.read_discriminant(place.into())?.0;
                 this.write_scalar(Scalar::from_uint(discr_val, dest.layout.size), dest)?;
             }
+            "is_const_eval" => {
+                this.write_scalar(Scalar::from_u8(1), dest)?;
```

Comment:
This is correct from the what we can do perspective, but that means the naming of the intrinsic is wrong.

Reply:
Idk anything better though

Reply:
In C++ this is called `is_constant_evaluated`.

Reply:
Will this code even be executed? We usually prefer the librustc_mir implementation of an intrinsic, and only fall back here when none exists there.

Reply:
When I started working on this, I started here and not in rust-lang/rust because I thought that rust-lang/rust would call this library to evaluate the intrinsics. Now I am not sure if this makes sense.

Reply:
Ah I see. No, that's not how the code is organized. The code in this repo is *only* used by Miri-the-tool; `rustc` (and its test suite) to not use the code here at all.